pennylane/interfaces/tf.py

```diff
@@ -138,6 +138,11 @@ def _execute(self, params, **input_kwargs):
         res = self.execute_device(args, input_kwargs["device"])
         self.set_parameters(all_params, trainable_only=False)
 
+        # save state
+        jac_options = getattr(self, "jacobian_options", dict())
+        if jac_options.get("cache_state", False):
+            state = input_kwargs["device"].state
+

```

Comment:
```suggestion
        # tape might not be a jacobian tape
        jac_options = getattr(self, "jacobian_options", dict())
        # cache state for adjoint jacobian computation
        if jac_options.get("cache_state", False):
            state = input_kwargs["device"].state

```
To match the torch style